### DISCUSSION TITLE
how to pass on a keybind to the shell

### DISCUSSION THREAD

**Discussion by d-r-a-b** [wezterm/wezterm#3530] [Q&A] [answered]
things like tmux are cognizant of the fact that you might want to send their keybinds on to the programs actually running in the shell. Eg. tmux uses `ctrl-b` to prefix many commands and so it intercepts this combination before it reaches a a program, and `ctrl-b ctrl-b` will allow you to send a `ctrl-b` to the actual program

Wezterm provides many default mappings, which seem to have been chosen with care to avoid stepping as much as possible stepping on common mappings. However, there are many mappings and I can forsee a future where I will need to send some combo to a program. after going through the documentation I wasn't able to find if there was a generic way to send a Wezterm-bound key combo into the shell programs, ideally through some prefix combo.

This easily might be described in the docs, they are extensive, but I couldn't find it. I can probably make this work with a key table, but then I have to update the "escaping" key table anytime my config or the default bindings get added to. This seems like a generic enough feature that I feel like it probably exists somehow and I just couldn't find it.

**Comment by wez** [comment, marked as answer]
Take a look at: [SendKey](https://wezfurlong.org/wezterm/config/lua/keyassignment/SendKey.html)
There isn't a generic, built-in, leader/key table way of "quoting" an arbitrary key press, but you could generate a key table for that with a lua for-loop and have the action use `SendKey` accordingly.

**Reply by d-r-a-b** [reply]
thanks for the quick reply and reminder that the wezterm config is really a Lua script. much thanks!
